89920 Add upgrade CLI for diff providers (#4508)
mesosphere/dcos-docs-site: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md

<p class="message--warning"><strong>IMPORTANT:</strong>Ensure your <code>dkp</code> configuration references the management cluster where you want to run the upgrade by setting the <code>KUBECONFIG</code> environment variable, or using the <code>--kubeconfig</code> flag, <a href="https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/">in accordance with Kubernetes conventions</a>.

> Upgrade the core addons in a cluster using the 'dkp upgrade addons' command specifying the cluster infrastructure (choose [aws, azure, vsphere, gcp, eks, preprovisioned]) and the name of the cluster.

Reviewer: ```suggestion
Upgrade the core addons in a cluster using the `dkp upgrade addons` command specifying the cluster infrastructure (choose `[aws, azure, vsphere, gcp, eks, preprovisioned]`) and the name of the cluster.
```

    - If an AMI was specified when initially creating a cluster for AWS, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-ami/">Konvoy Image Builder</a>.
    - If an Azure Machine Image was specified for Azure, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-azure-image/">Konvoy Image Builder</a>.

> 2. Upgrade the Kubernetes version of the control plane. As shown in update addons section above, replace infrastructure provider name in the command [aws, azure, vsphere, gcp, eks, preprovisioned].

Reviewer: ```suggestion
2. Upgrade the Kubernetes version of the control plane. As shown in update addons section above, replace infrastructure provider name in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

```bash
dkp update controlplane aws --cluster-name=${CLUSTER_NAME} --kubernetes-version=v1.22.8
```

The output should be similar to the below example with the provider name corresponding to the one you entered in the command line:

```text
Updating control plane resource controlplane.cluster.x-k8s.io/v1beta1, Kind=KubeadmControlPlane default/my-aws-cluster-control-plane
Waiting for control plane update to finish.
 ✓ Updating the control plane
```
    

> 3. Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool. Also, replace the name of the infrastructure provider in the command [aws, azure, vsphere, gcp, eks, preprovisioned].

Reviewer: ```suggestion
3. Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool. Also, replace the name of the infrastructure provider in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

[backup]: ../../../../../kommander/2.3/backup-and-restore#back-up-on-demand
[envariables2]: ../../../choose-infrastructure/aws/quick-start-aws#configure-aws-prerequisites
[envariables3]: ../../../choose-infrastructure/vsphere/new#create-a-new-vsphere-kubernetes-cluster

> [envariables4]: ../../../choose-infrastructure/eks/quick-start-eks#configure-eks-prerequisites

Reviewer: ```suggestion
[envariables4]: ../../../choose-infrastructure/eks/quick-start#configure-eks-prerequisites
```
